File: readability.py
from cs50 import get_string

text = get_string("Text: ")
letters = 0
sentences = 0
words = 1

for i in range(len(text)):
    if ord(text[i]) >= 65 and ord(text[i]) <= 122:
        letters += 1
    elif text[i] == '.' or text[i] == '?' or text[i] == '!':
        sentences += 1
    elif text[i] == ' ' or text[i] == '"' and ord(text[i + 1]) >= 65 and ord(text[i + 1]) <= 122:
        words += 1
L = (letters / words) * 100
S = (sentences / words) * 100
x = (0.0588 * L)
y = (0.296 * S)

readability = x - y - 15.8

if readability > 16:
    print("Grade 16+")
elif readability < 1:
    print("Before Grade 1")
else:
    print("Grade", round(readability))

File: readability.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    int toascii(int c);
    string text = get_string("Text: ");
    int letters = 0;
    int sentences = 0;
    int words = 1;
    for (int i = 0; i < strlen(text); i++)
    {
        if (toascii(text[i]) >= 65 && toascii(text[i]) <= 122)
        {
            letters++;
        }
        if ((text[i] == '.' || text[i] == '?' || text[i] == '!'))
        {
            sentences++;
        }
        else if (text[i] == ' ' && toascii(text[i + 1]) >= 65 && toascii(text[i + 1]) <= 122)
        {
            words++;
        }
    }
    float L = ((float) letters / (float) words) * 100;
    float S = ((float) sentences / (float) words) * 100;
    float x = (0.0588 * ((float) L));
    float y = (0.296 * ((float) S));
    double readability = (float) x - (float) y - 15.8;
    if (readability > 16)
    {
        printf("Grade 16+\n");
    }
    else if (readability < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %.0f\n", readability);
    }
}